src/providers/krb5/krb5_child.c

```diff
@@ -2930,12 +2928,47 @@ static errno_t renew_tgt_child(struct krb5_req *kr)
         return ERR_INVALID_CRED_TYPE;
     }
 
+    DEBUG(SSSDBG_TRACE_LIBS, "Renewing a ticket '%s'\n", ccname);
+
     kerr = krb5_cc_resolve(kr->ctx, ccname, &ccache);
     if (kerr != 0) {
         KRB5_CHILD_DEBUG(SSSDBG_CRIT_FAILURE, kerr);
         goto done;
     }
 
+    if (strncmp(ccname, "FILE:", 5) == 0) {
+        /* This might be coming from `init_renew_tgt()->check_ccache_files()`.
+         * Let's make sure those are still renewable.
+         */
+        krb5_creds mcred;
+        krb5_creds old_cred;
+        time_t renew_till;
+        time_t endtime;
+        time_t now = time(NULL);
+        memset(&mcred, 0, sizeof(mcred));
+        memset(&old_cred, 0, sizeof(old_cred));
+        mcred.client = kr->princ;
+
+        kerr = krb5_cc_retrieve_cred(kr->ctx, ccache, 0, &mcred, &old_cred);
```

Comment:
Hi,

there already is `get_tgt_times()`. Maybe this can be modified to accept a `krb5_ccache` as an alternative to the `ccanme`? This wouldn't make the patch smaller, but the code here might be easier to read/understand?

bye,
Sumit

Reply:
Done.